### PR TITLE
clock: add NewTimer

### PR DESCRIPTION
**Pull request by rogpeppe**
This allows mock clocks to use timers in the usual way.
This is associated with a companion branch in juju
testing (https://github.com/juju/testing/pull/108) which needs
this as a dependency update.

### CODE REVIEW

File: clock/wall.go [hunk 1]
```diff
@@ -25,5 +25,17 @@ func (wallClock) After(d time.Duration) <-chan time.Time {
 }
 
 func (wallClock) AfterFunc(d time.Duration, f func()) Timer {
-	return time.AfterFunc(d, f)
+	return wallTimer{time.AfterFunc(d, f)}
+}
+
+func (wallClock) NewTimer(d time.Duration) Timer {
```
Comment (alesstimec) [inline clock/wall.go]: godoc, please

File: clock/wall.go [hunk 2]
```diff
@@ -25,5 +25,17 @@ func (wallClock) After(d time.Duration) <-chan time.Time {
...
+	return wallTimer{time.NewTimer(d)}
+}
+
+type wallTimer struct {
+	*time.Timer
+}
+
+func (t wallTimer) Chan() <-chan time.Time {
```
Comment (alesstimec) [inline clock/wall.go]: godoc please

File: clock/wall.go [hunk 3]
```diff
@@ -25,5 +25,17 @@ func (wallClock) After(d time.Duration) <-chan time.Time {
 }
 
 func (wallClock) AfterFunc(d time.Duration, f func()) Timer {
```
Comment (alesstimec) [inline clock/wall.go]: godoc, please

File: clock/wall.go (hunk 1 above)
Comment (rogpeppe) [inline clock/wall.go]: done

File: clock/wall.go (hunk 2 above)
Comment (rogpeppe) [inline clock/wall.go]: done

**Comment by rogpeppe** [comment]
$$merge$$

**Comment by rogpeppe** [comment]
$$merge$$

**Pull request merged**